src/bus/mod.rs

```diff
@@ -46,30 +46,40 @@ enum Listener {
 
 impl Bus {
     pub async fn for_address(address: Option<&str>) -> Result<Self> {
-        let mut address = match address {
+        let address = match address {
             Some(address) => Address::from_str(address)?,
             None => Address::from_str(&default_address())?,
         };
-        let guid: OwnedGuid = match address.guid() {
-            Some(guid) => guid.to_owned().into(),
-            None => {
-                let guid = Guid::generate();
-                address = address.set_guid(guid.clone())?;
-
-                guid.into()
-            }
+        let address = if address.guid().is_some() {
+            address
+        } else {
+            let guid = Guid::generate();
+            address.set_guid(guid.clone())?
+        };
```

Comment:
same here. Please don't switch from match to if. I find the former more readable, despite the extra level of indentation.